### DISCUSSION TITLE
how to set up local config with limitations of source?

### DISCUSSION THREAD

**Discussion by ExpandingMan** [nushell/nushell#14064] [Q&A] [answered]
In other shells I always have a "local" config file that is not committed to my nushell config repository, but because `source` is evaluated at parse time, I can't figure out a way to do this with nushell.  Conditional sourcing does not work, as is well known, but also conditionally creating a blank file does not work because `source` is evaluated before.  Has anyone figured out a way to do this?

**Comment by fdncred** [comment]
I think this approach should work. I'm pretty sure you can construct paths with the `path` family of command and with the `$nu` variables because they're all cost too.
```nushell
const path1 = /path/to/config1.nu
const path2 = /path/to/config2.nu
source (if condition { $path1 } else { $path2 })
```

**Reply by ExpandingMan** [reply]
This is very weird, I can't make sense out of it.  I have
```nu
const _local_config_null_path = '.localnull.nu'
const _local_config_path = 'local.nu'
if ($_local_config_path | path exists) {echo 'here!'}
source (if ($_local_config_path | path exists) {$_local_config_path} else {$_local_config_null_path})
```
It seems like the `else` clause here *always* evaluates, it never evaluates the first clause.  The first `if` statement never seems to trigger regardless whether I condition it on `_local_config_path` or `_local_config_null_path`.  I have created *both* those local files.

**Reply by weirdan** [reply]
try extracting the conditions:
```nushell
const local_exists = $_local_config_path | path exists
const file_to_source = if $local_exists { $_local_config_path } else { $_local_config_null_path }
source $file_to_source
```

I'm pretty sure this worked for me several weeks ago.

**Comment by ExpandingMan** [comment]
```nu
const _local_exists = 'local.nu' | path exists
const _local_source = if $_local_exists { 'local.nu' } else { '.local.nu' }
source $_local_source
```
Still seems to always evaluate `_local_source` to `.local.nu` regardless of whether the other one exists.

**Reply by weirdan** [reply, marked as answer]
I've added the following to my config:
```nushell
const _local_exists = $nu.default-config-dir ++ "/local.nu" | path exists
const _local_source = if $_local_exists { $nu.default-config-dir ++ "/local.nu" } else { $nu.default-config-dir ++ "/.local.nu" }
source $_local_source
```
Then I created two files, `~/.config/nushell/local.nu` and `~/.config/nushell/.local.nu` with simple echo statements.

Then:

![image](https://github.com/user-attachments/assets/a2085c91-c19b-4ddb-a6d1-e9230fd888f6)

This is on nu 0.98

**Reply by ExpandingMan** [reply]
Ah, ok so I guess it was some kind of path confusion... *really* do not understand why it was able to find one file but not the other, but at least it works now, thanks!

**Reply by fdncred** [reply]
It was because your condition was wrong. Nushell doesn't allow you to do `if some_file` because `some_file` does not return a bool.

**Reply by ExpandingMan** [reply]
I had `path exists`, see above.

**Reply by fdncred** [reply]
originally you didn't. I missed where you changed it. But that change didn't have the full path on the files. I wouldn't think that should work unless those files are in your NU_LIB_DIRS folder(s).
